src/main/java/duke/DukeTaskList.java

```diff
@@ -0,0 +1,178 @@
+package duke;
+
+import duke.task.*;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Encapsulates the list in Duke that stores all the tasks
+ */
+public class DukeTaskList {
+    public List<Task> taskList;
+
+    /**
+     * Constructor for the DukeTaskList class.
+     */
+    public DukeTaskList() {
+        taskList = new ArrayList<>(100);
+    }
+
+    /**
+     * Method to read data from the save file, and load them
+     * into the list in a current run of Duke.
+     *
+     * @param type  type of task stored in save file.
+     * @param state state of task stored in save file. 0 for not done and 1 for done.
+     * @param body  includes the description, date(if any) and time(if any)
+     *              stored in save file.
+     */
+    public void loadDataToList(String type, String state, String body) {
+        Task currTask = null;
+        String[] bodySplit = body.split("\\|", 4);
+
+        switch (type) {
+        case "T":
+            currTask = new ToDos(body);
+            break;
+        case "D":
+            currTask = new Deadlines(bodySplit[0],
+                LocalDate.parse(bodySplit[1]),
```

Comment:
Should the indentation for this wrapped line be 8 spaces more than the parent line instead? I think it is currently only 4 spaces.

I have noticed this in other parts of the code as well.